llvm/lib/DebugInfo/DWARF/DWARFDebugLine.cpp

```diff
@@ -857,7 +860,7 @@ Error DWARFDebugLine::LineTable::parse(
       }
       if (OS)
         State.Row.dump(*OS);
-      State.appendRowToMatrix();
+      State.appendRowToMatrix(LineTableSeqOffset);
```

Comment:
When ever we create a new Sequenced within `State` we can update the line offset in the current sequence. We can force this by having the `Seqeuence` constructor take an offset if possible and then the offset can be const. Or we can just do something like:
```
State.SetSequenceOffset(uint64_t Offset);
```
And call this before entering the while look and after we get a `DW_LNE_end_sequence`